### PR TITLE
fixes #4869 feat(nimbus): expose timeout data in V5 API

### LINKED ISSUE #4869: Expose timeout data in V5 API
To support the new timeout UI we need to look up the timeout in the changelogs and expose it in a convenient field in the V5 API.

We should

* Add a manager method to NimbusChangeLog to look up the most recent timeout changelog by finding the changelog which moves from publish_status Approved>Idle
* Add a field in the V5 NimbusExperiment type called `timeout` that is None or includes the entire changelog object so the frontend can display the date of the timeout

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/EXP-1096)
┆Issue Number: EXP-1096

### PR DESCRIPTION
**Pull request by lmorchard**
Because:

- To support the new timeout UI we need to look up the timeout in the
  changelogs and expose it in a convenient field in the V5 API.

This commit:

- Adds a `timeout` property to the NimbusExperiment model that looks up
  the latest changelog entry that transitions from publish status waiting
  to review, as long as the experiment is still in review status

- Exposes the `timeout` model property as a field in GQL queries

### CODE REVIEW

File: app/experimenter/experiments/models/nimbus.py
Comment (lmorchard) [inline app/experimenter/experiments/models/nimbus.py:175]: I might be over-thinking this. But, I figured we'd want some more logic here:

* Only report a timeout if the experiment has gotten kicked from WAITING back to REVIEW - and is *still* in REVIEW publishing status.
* The timeout reported is None if/when the experiment gets another attempt going from REVIEW -> APPROVED -> WAITING

File: app/experimenter/experiments/tests/test_models/test_models_nimbus.py
Comment (lmorchard) [inline app/experimenter/experiments/tests/test_models/test_models_nimbus.py:462]: I *think* all the case we wanted to cover here are within DRAFT and LIVE experiment statuses

**Review by lmorchard — COMMENTED** [review]
Dropped some notes on some further thinking I did around logic here

**Review by jaredlockhart — COMMENTED** [review]
Thanks for hopping on this @lmorchard !  I'd suggest waiting until https://github.com/mozilla/experimenter/pull/4896 lands, rebasing, and following the same pattern (ie putting the lookup method in the NimbusChangeLogManager)

**Comment by lmorchard** [comment]
Hmm, if I move from an experiment model property to a changelog manager method, I think that would always give us the last recent timeout but have to shift the decision on whether the timeout is currently relevant to the frontend (e.g. is the experiment timed out or waiting for another approval)

**Comment by lmorchard** [comment]
Rebased, reworked to follow in the footsteps of PR #4896

**Review by jaredlockhart — APPROVED** [review]
Awesome looks right to me!

**Pull request merged**
